src/AzureAuth/CommandMain.cs

```diff
@@ -218,6 +222,47 @@ public static string PrefixedPromptHint(string promptHint)
             return $"{PromptHintPrefix}: {promptHint}";
         }
 
+        /// <summary>
+        /// Generates event data from the AuthFlowResult.
+        /// </summary>
+        /// <param name="result">The AuthFlowResult.</param>
+        /// <returns>The event data.</returns>
+        public EventData GenerateEventData(AuthFlowResult result)
+        {
+            if (result == null)
+            {
+                return null;
+            }
+
+            var eventData = new EventData();
+            eventData.Add("authflow", result.AuthFlowName);
+            eventData.Add("success", result.Success);
+
+            var correlationIDs = new List<string>();
+
+            if (result.Errors.Any())
+            {
+                var error_messages = JsonSerializer.Serialize(result.Errors.Select(error => ExceptionsExtensions.ToFormattedString(error)));

```

Comment:
Why are we serializing this since `ExceptionsExtensions.ToFormattedString(error)` does return a string?

Reply:
I changed it to use the previous method which takes the exceptions messages and converts them to line separated messages.